### ISSUE TITLE
bug: `ion-datetime` time picker incorrectly positioned when rapidly opened or in React/Vue inline ion-popover

### ISSUE THREAD

**Issue by amiraiouaz** [ionic-team/ionic-framework#24415]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

Hello,

First of all, I want to thank you for this great framework.

I just started an ionic project with the version 6.x (with **Vue.js**) and I am having a little issue with the time popover when I use the ion-datetime in a popover:

![image](https://user-images.githubusercontent.com/22788646/146205670-b71d8709-d665-4230-a224-83fe734ccecf.png)

This is my code:

```html
<ion-item
    id="datepicker"
    button
    detail="false"
  >
    <ion-icon
      slot="end"
      :icon="calendarOutline"
    />
    <ion-popover
      trigger="datepicker"
      size="cover"
      show-backdrop="false"
      translucent
    >
      <ion-datetime
        first-day-of-week="1"
        hour-cycle="h23"
        size="cover"
        show-default-time-label="false"
        :presentation="presentation"
        :locale="locale"
      />
    </ion-popover>
</ion-item>
```

### Expected Behavior

This should be the expected position:

<img src="https://user-images.githubusercontent.com/22788646/146206684-14db0e00-2b70-4fd6-8d16-1e81c1f57975.png" width="350px" />

### Steps to Reproduce

- Click on ion-item
- Click on time button
- Position is not correct

I am doing something wrong or is this a bug?

Thanks a lot in advance for your help.

### Code Reproduction URL

- https://github.com/sean-perkins/datetime-gh-24415

### Ionic Info

My package.json:

```json
  "dependencies": {
    "@capacitor/app": "1.0.7",
    "@capacitor/core": "3.3.3",
    "@capacitor/haptics": "1.1.3",
    "@capacitor/keyboard": "1.2.0",
    "@capacitor/status-bar": "1.0.6",
    "@capacitor/storage": "^1.2.3",
    "@ionic/vue": "^6.0.0",
    "@ionic/vue-router": "^6.0.0",
    "core-js": "^3.6.5",
    "vue": "^3.2.21",
    "vue-router": "^4.0.12"
  }
```

**Comment by sean-perkins** [comment]
Thanks for reporting this issue! I've been able to reproduce the behavior here: https://github.com/sean-perkins/datetime-gh-24415.

On an initial look, I believe it's a side effect of nested popover usages. We use popover internally for the time picker and popovers are position fixed by default. 

We'll track this issue as a bug.

**Comment by sean-perkins** [comment]
This issue is only present in the React and Vue implementations. I believe it's a custom elements issue. Will share more soon.

**Comment by creatxrgithub** [comment]
time picker position with web browser

`import React, { useState, useRef } from 'react';
import { IonContent, IonHeader, IonPage, IonTitle, IonToolbar, IonDatetime } from '@ionic/react';
import ExploreContainer from '../components/ExploreContainer';
import './Tab1.css';

const Tab1: React.FC = () => {
  const [selectedDate, setSelectedDate] = useState('');
 
  return (
    <IonPage>
      <IonHeader>
        <IonToolbar>
          <IonTitle>Tab 1</IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent fullscreen>
        <IonHeader collapse="condense">
          <IonToolbar>
            <IonTitle size="large">Tab 1</IonTitle>
          </IonToolbar>
        </IonHeader>
        <IonDatetime hour-cycle="h23" locale="en-GB" value={selectedDate} onIonChange={e => setSelectedDate(e.detail.value!)}></IonDatetime>

      </IonContent>
    </IonPage>
  );
};

export default Tab1;
`

![time_picker_position_problem](https://user-images.githubusercontent.com/27437178/147032886-75769399-be1a-4da9-8434-5b73f1f6eb5f.png)

**Comment by mixalbl4-127** [comment]
https://github.com/ionic-team/ionic-framework/pull/24532
Possible fix (works for me)
@liamdebeasi

**Comment by liamdebeasi** [comment]
Hi everyone,

This issue was was fixed in Ionic 6.0.4 but we missed closing out the issue. Please update your apps to the latest version of Ionic to receive this fix. Thanks!
